compiler/src/dotty/tools/dotc/core/unpickleScala2/Scala2Unpickler.scala

```diff
@@ -454,6 +454,19 @@ class Scala2Unpickler(bytes: Array[Byte], classRoot: ClassDenotation, moduleClas
       // skip this member
       return NoSymbol
 
+    // Skip case accessor `<xyz>$access$<idx>` and make `<xyz>` the case accessor
+    if flags.is(CaseAccessor) && name.toString().contains("$access$") then
+      // The only way to know if the accessor needs the additional `CaseAccessor` flag is
+      // when we see the `<xyz>$access$<idx>` accessor. When we load `<xyz>` we have not
+      // loaded or have skipped `<xyz>$access$<idx>`. Furthermore, its flags are
+      // undistinguishable from non-case parameter accessors (example
+      // `case class Foo(private[p] var x: T)(private[p] var y: T)`).
+      val accessorIndex = name.toString().split('$').last.toInt
+      val accessors = owner.asClass.info.decls
```

Comment:
This is too dangerous at this point. Store the necessary information in a map like `paramsOfMethodType` and clean up when completing the class decls.

Reply:
Cleaning up when completing the symbol did not work https://github.com/lampepfl/dotty/pull/18907#discussion_r1396815766.

The other option I found is to use `symScope(owner)`. This is also used above when skipping other symbols.